components/style/gecko/wrapper.rs

```diff
@@ -1656,6 +1657,24 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
         node_info.mInner.mNamespaceID == (structs::root::kNameSpaceID_XHTML as i32) &&
         node.owner_doc().mType == structs::root::nsIDocument_Type::eHTML
     }
+
+    fn blocks_ancestor_combinators(&self) -> bool {
+        use gecko_bindings::structs::NODE_IS_ANONYMOUS_ROOT;
+        if self.flags() & (NODE_IS_ANONYMOUS_ROOT as u32) == 0 {
+            return false
+        }
+
+        match self.parent_element() {
+            Some(e) => {
+                // If this element is the shadow root of an use-element shadow
+                // tree, according to the spec, we should not match rules
+                // cross the shadow DOM boundary.
+                e.get_local_name().as_ptr() == atom!("use").as_ptr() &&
```

Comment:
e is a GeckoElement, get_local_name returns &WeakAtom, and type of &local_name!("use") is &Atom, so we will get a compile error like this:
 1669 |                 e.get_local_name() == &local_name!("use")
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::cmp::PartialEq<gecko_string_cache::Atom>` is not implemented for `gecko_string_cache::WeakAtom`